src/sentry/api/paginator.py

```diff
@@ -746,3 +750,84 @@ def get_result(self, limit=100, cursor=None):
             results = self.on_results(results)
 
         return CursorResult(results=results, next=next_cursor, prev=prev_cursor)
+
+
+class SnubaRequestPaginator:
+    def __init__(
+        self,
+        query: Query,
+        dataset: str,
+        app_id: str,
+        tenant_ids: dict,
+        order_by: Optional[str] = None,
+        limit: Optional[int] = None,
+        on_results: Optional[Callable[[Sequence[Any]], Any]] = None,
+    ):
+        self.query = query
+        if dataset not in {ds.value for ds in Dataset}:
+            raise ValueError(f"Dataset {dataset} does not exist")
+        self.dataset = dataset
+        self.app_id = app_id
+        self.tenant_ids = tenant_ids
+        if order_by:
```

Comment:
Nit: I think that if no `order_by` is provided and there's none on the query we should error. Pagination won't work too well without a fixed order